src/borg/nonces.py

```diff
@@ -64,24 +64,20 @@ def ensure_reservation(self, nonce_space_needed):
 
         if self.end_of_nonce_reservation:
             # we already got a reservation, if nonce_space_needed still fits everything is ok
-            next_nonce = int.from_bytes(self.enc_cipher.iv, byteorder='big')
+            next_nonce_bytes = self.cipher.next_iv()
+            next_nonce = int.from_bytes(next_nonce_bytes, byteorder='big')
             assert next_nonce <= self.end_of_nonce_reservation
             if next_nonce + nonce_space_needed <= self.end_of_nonce_reservation:
+                self.cipher.set_iv(next_nonce_bytes)
```

Comment:
i’m not sure i like setting the iv here. Now that iv is very much more explicit i wonder if next_iv should be a parameter and next_nonce_bytes the return of this method?